prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -123,14 +123,17 @@ class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLev
     }
 
     /**
-     * Check BOTH authentication and authorization for this [request], and if both succeed, execute [block]
-     * [oktaSender] is true if we expect this call comes from a reportstream Sender named [organizationName]
+     * Check BOTH authentication and authorization for this [request], and if both succeed, execute [block].
+     * When an [organizationName] is provided, this allows access at the User principal level for users with a
+     * claim to that [organizationName].
+     * If [requireSenderClaim] is true, then only allow access if that organization is a Sender org.
+     * If [requireSenderClaim] is false, then both Sender orgs and non-Sender orgs are allowed access.
      * @return a suitable HttpResponse.
      */
     fun checkAccess(
         request: HttpRequestMessage<String?>,
         organizationName: String = "",
-        oktaSender: Boolean = false,
+        requireSenderClaim: Boolean = false,
```

Comment:
changed the name of this variable to clarify its purpose.  Also attempted to document it ^^^